bigquery/google/cloud/bigquery/job.py

```diff
@@ -969,62 +974,127 @@ def from_api_repr(cls, resource, client):
         return job
 
 
-class _ExtractConfiguration(object):
-    """User-settable configuration options for extract jobs.
+class ExtractJobConfig(object):
+    """Configuration options for extract jobs.
 
-    Values which are ``None`` -> server defaults.
+    All properties in this class are optional. Values which are ``None`` ->
+    server defaults.
     """
-    _compression = None
-    _destination_format = None
-    _field_delimiter = None
-    _print_header = None
+
+    def __init__(self):
+        self._properties = {}
+
+    compression = Compression('compression', 'compression')
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.extract.compression
+    """
+
+    destination_format = DestinationFormat(
+        'destination_format', 'destinationFormat')
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.extract.destinationFormat
+    """
+
+    field_delimiter = _TypedApiResourceProperty(
+        'field_delimiter', 'fieldDelimiter', six.string_types)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.extract.fieldDelimiter
+    """
+
+    print_header = _TypedApiResourceProperty(
+        'print_header', 'printHeader', bool)
+    """See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.extract.printHeader
+    """
+
+    def to_api_repr(self):
+        """Build an API representation of the extact job config.
+
+        :rtype: dict
+        :returns: A dictionary in the format used by the BigQuery API.
+        """
+        resource = copy.deepcopy(self._properties)
```

Comment:
No need to assign to a local variable, is there?

Reply:
Removed.